src/EventStore.Core/Services/ElectionsService.cs

```diff
@@ -104,11 +107,18 @@ public class ElectionsService : IHandle<SystemMessage.BecomeShuttingDown>,
 			_clusterSize = clusterSize;
 			_writerCheckpoint = writerCheckpoint;
 			_chaserCheckpoint = chaserCheckpoint;
+			_proposalCheckpoint = proposalCheckpoint;
 			_epochManager = epochManager;
 			_getLastCommitPosition = getLastCommitPosition;
 			_nodePriority = nodePriority;
 			_timeProvider = timeProvider;
 
+			var lastEpoch = _epochManager.LastEpochNumber;
+			if (_proposalCheckpoint.Read() < lastEpoch) {
```

Comment:
_proposalCheckpoint.Read() can be replaced with _proposalCheckpoint.ReadNonFlushed() everywhere since the value is cached in memory.